semgrep/semgrep/evaluation.py

```
@@ -172,129 +172,104 @@ def _evaluate_single_expression(
     flags: Optional[Dict[str, Any]] = None,
 ) -> Set[Range]:
 
-    assert expression.pattern_id, f"<internal error: expected pattern id: {expression}>"
```

`assert`s are a no-op most of the time (e.g. when using `-OO` for optimization, which Python does when building wheels). The below `if` check should cover this :+1:

---

I think the below if will silently continue if the pattern_id is falsy? (which i guess is the same behavior as an optimized out assert) Do we want to make this noisier or is it fine to continue?

---

I'll convert it to a `raise SemgrepError` to mimic the previous behavior :+1: